uaclient/cli.py

```diff
@@ -1340,17 +1343,6 @@ def action_auto_attach(args, *, cfg):
             messages.UNABLE_TO_DETERMINE_CLOUD_TYPE
         )
 
-    current_iid = identity.get_instance_id()
-    if cfg.is_attached:
-        prev_iid = cfg.read_cache("instance-id")
-        if str(current_iid) == str(prev_iid):
-            raise exceptions.AlreadyAttachedOnPROError(str(current_iid))
-        print("Re-attaching Ubuntu Pro subscription on new instance")
-        if _detach(cfg, assume_yes=True) != 0:
-            raise exceptions.UserFacingError(
-                messages.DETACH_AUTOMATION_FAILURE
-            )
-
```

Comment:
I think this is reasonable given the change in the contract server hanlding which disambiguates multiple attached machines under the same contract. +1